### PR TITLE
ci: add generate for mocks

### PR DESCRIPTION
**Pull request by stevenh**
Add generation for mocks, which can be run using make generate or go generate ./...

Correct output-format for golangci-lint as githubs-actions has been replaced by colored-line-numbers.

Add pre-commit target to help check code is valid before pushing.

Implements: #2765

### CODE REVIEW

File: .github/workflows/ci-test-go.yml [hunk 1]
```diff
@@ -85,6 +85,10 @@ jobs:
           #           takes precedence over all other caching options.
           skip-cache: true
 
+      - name: generate
+        working-directory: ./${{ inputs.project-directory }}
+        run: make generate
```
Comment (mdelapenya) [inline .github/workflows/ci-test-go.yml:90]: I think we want to verify there are no modified files after this step

**Review by mdelapenya — COMMENTED** [review]
LGTM, just a comment regarding checking if there are modified files on CI

File: .github/workflows/ci-test-go.yml (hunk 1 above)
Comment (stevenh) [reply .github/workflows/ci-test-go.yml]: Yep, saw you had mod tidy in there and assumed it was doing that, but on checking it's not, so added to both.

File: .github/workflows/ci-test-go.yml [hunk 2]
```diff
@@ -85,13 +85,23 @@ jobs:
...
         working-directory: ./${{ inputs.project-directory }}
         run: go mod verify
 
       - name: modTidy
         working-directory: ./${{ inputs.project-directory }}
-        run: make tidy
+        shell: bash
+        run: |
+          make tidy
+          git --no-pager diff && [[ 0 -eq $(git status --porcelain | wc -l) ]]
```
Comment (mdelapenya) [inline .github/workflows/ci-test-go.yml:104]: Neat! This will avoid any mistakes on CI. Thanks!

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

Waiting for the CI to merge.

As a follow-up, we are going to replace the custom mocks from the wait strategies with this approach, right?

**Review by mdelapenya — DISMISSED** [review]

**Comment by stevenh** [comment]
> LGTM, thanks!
> 
> Waiting for the CI to merge.
> 
> As a follow-up, we are going to replace the custom mocks from the wait strategies with this approach, right?

Yep already have some [work in progress here](https://github.com/testcontainers/testcontainers-go/pull/2774)

**Comment by stevenh** [comment]
Unrelated failure, should be good now @mdelapenya

**Review by mdelapenya — APPROVED** [review]

**Pull request merged**